ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/transaction/arjunacore/jca/XATerminatorImple.java

```diff
@@ -241,6 +241,12 @@ public int prepare (Xid xid) throws XAException
 
             switch (tx.doPrepare())
             {
+            case TwoPhaseOutcome.PREPARE_ONE_PHASE_COMMITTED:
+                // Should never happen Would be great if there was heuristic commit :(
+                SubordinationManager.getTransactionImporter()
+                        .removeImportedTransaction(xid);
+                jtaLogger.i18NLogger.fatalSubordinate1PCDuringPrepare(xid);
+                throw new XAException(XAException.XAER_RMERR);
```

Comment:
I updated the description with some wording to describe this. There are two resources. If you look at the test case you can see what I am protecting against, if you removed the assignment to false of dynamic1PC from SubordinateAtomicAction  then it wouldn't pass actually